### PR TITLE
HODC added ‘SPA_WebAPI_EF’ application.

### PR DESCRIPTION
**Pull request by SandeepNayak077**
1) Implemented code to perform INSERT/UPDATE/DELETE operation in Entity Framework
2) Did UI improvement of SampleEF screen
3) Did code review and code refractration

### CODE REVIEW

**Comment by daisukenishino2** [comment]
Note just in case.
Written information in the original repository is lost when you re-fork.
Therefore, when changes has not been merged in original repository, then do not re-fork.

**Comment by daisukenishino** [comment]
I filled out below as notes. Please check.
https://github.com/OpenTouryoProject/OpenTouryo/blob/develop/CONTRIBUTING.md

**Comment by SandeepNayak077** [comment]
Thank you Nishino-san for the valuable information about Github.
HODC will take care while doing re-fork.

**Comment by daisukenishino** [comment]
You will need to reflect the following Issue on 'SPA_WebAPI_EF'. Please check.

- Technology Research of Entity Framework.  
https://github.com/OpenTouryoProject/SampleProgram/issues/3
- Technology Research of Entity Framework2  
https://github.com/OpenTouryoProject/SampleProgram/issues/6
- We will accept your proposal for UI improvement of SPA.  
https://github.com/OpenTouryoProject/SampleProgram/issues/5

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Test.viewmodel.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿function TestViewModel() {
+	// Shipper テーブルの各項目
+	this.ShipperId = ko.observable("Shipper1");
+	this.CompanyName = ko.observable("Japan");
+	this.Phone = ko.observable("052-569-2597");
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Test.viewmodel.js]: This is a phone number to exist in Japan. Please do not use the real phone number.
Please revise it to an imaginary phone number (e.g.,: 012-345-6789).

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Codes/Dao/DaoLayer.cs [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+
+        /// <summary>
+        /// Select records from shippers table based on various conditions
+        /// </summary>
+        /// <param name="testParameter"></param>
+        /// <param name="testReturn"></param>
+        public void Select(TestParameterValue testParameter, TestReturnValue testReturn)
+        {
+            // Create Shipper data table using entity model
+            DataTable dt = new DataTable();
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Codes/Dao/DaoLayer.cs]: Please assume the return value List<Shipper>. (not DataTable)

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Sample.viewmodel.js [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+
+    // 通信制御
+    this.ddlTransmissionItems = ko.observableArray([
+        { displayText: "Webサービス呼出", value: "testWebService" },
+        { displayText: "インプロセス呼出", value: "testInProcess" }
+    ]);
+    this.ddlTransmission = ko.observable("testWebService");
+
+    // Shipper テーブルの各項目
+    this.ShipperId = ko.observable("");
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Sample.viewmodel.js]: Please unify the naming methods of the variable in either Camel method or Pascal method.

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Sample.viewmodel.js [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+    ]);
+    this.ddlOrderSequence = ko.observable("A");
+
+    // 処理結果 (正常系)
+    this.Result = ko.observable("");
+
+    // 処理結果 (異常系)
+    this.ErrorMessage = ko.observable("");
+
+    this.isVisibletblShippers = ko.observable(false);
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Sample.viewmodel.js]: Please make the top of the word a capital letter. (isVisible"T"blShippers)

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Sample.viewmodel.js [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+    this.isVisibletblCountShippers = ko.observable(false);
+    this.isVisibletblCrudOperation = ko.observable(false);
+    this.isVisibletblDataProvider = ko.observable(true);
+    this.flag = ko.observable(false);
+    this.flag1 = ko.observable(false);
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Sample.viewmodel.js]: We cannot understand the use of the variable named "flag" or "flag1".
Please attach the variable name that is easy to understand a use.

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Sample.viewmodel.js [hunk 6]
```diff
@@ -0,0 +1,556 @@
...
+    this.isMessage = ko.observable(false);
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Sample.viewmodel.js]: What is the use of variable "isMessage"?
If it is a variable to control visible / invisible, please name it "is'Visible'Message".

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Views/Shared/_AngularJSLayout.cshtml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+﻿<!DOCTYPE html>
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Views/Shared/_AngularJSLayout.cshtml]: This "SPA_WebAPI_EF" sample does not use this layout.
Therefore, please delete this layout.

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Content/SampleEF.css [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    border: solid 1px #000000;
+}
+
+.btn:hover {
+    background-color: #61618f;
+    height: 30px;
+    color: #fff;
+}
+
+.btn {
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Content/SampleEF.css]: If you want to define unified style on all buttons, we think that it is more simple to define following style to css file. (than to use "class")

input[type=button] {
  /* define style here */
}

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Sample2.viewmodel.js [hunk 9]
```diff
@@ -0,0 +1,505 @@
+﻿var Sample2ViewModel = function ($scope) {
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Sample2.viewmodel.js]: This sample does not use AngularJS.
Therefore, please delete this file.

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Test.viewmodel.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+﻿function TestViewModel() {
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/Test.viewmodel.js]: This file is not used in a sample.
Therefore, please delete this file.

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/todo.bindings.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
+﻿// Hooks up a form to jQuery Validation
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/todo.bindings.js]: This file is not used in a sample.
Therefore, please delete this file.

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/todo.datacontext.js [hunk 12]
```diff
@@ -0,0 +1,109 @@
+﻿window.todoApp = window.todoApp || {};
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/todo.datacontext.js]: This file is not used in a sample.
Therefore, please delete this file.

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/todo.model.js [hunk 13]
```diff
@@ -0,0 +1,78 @@
+﻿(function (ko, datacontext) {
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/todo.model.js]: This file is not used in a sample.
Therefore, please delete this file.

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/todo.viewmodel.js [hunk 14]
```diff
@@ -0,0 +1,44 @@
+﻿window.todoApp.todoListViewModel = (function (ko, datacontext) {
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/Scripts/app/todo.viewmodel.js]: This file is not used in a sample.
Therefore, please delete this file.

File: Template/SPA_WebAPI_EF/SPA_WebAPI_EF/App_Start/BundleConfig.cs [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+
+            // 開発と学習には、Modernizr の開発バージョンを使用します。次に、実稼働の準備が
+            // できたら、http://modernizr.com にあるビルド ツールを使用して、必要なテストのみを選択します。
+            bundles.Add(new ScriptBundle("~/bundles/modernizr").Include(
+                        "~/Scripts/modernizr-*"));
+
+            bundles.Add(new StyleBundle("~/Content/css").Include(
+                "~/Content/Site.css",
+                "~/Content/SampleEF.css",
+                "~/Content/TodoList.css"));
```
Comment (yusukemaegawa) [inline Template/SPA_WebAPI_EF/SPA_WebAPI_EF/App_Start/BundleConfig.cs]: The "TodoList.css" file does not exist.
Please delete this line.

**Comment by daisukenishino** [comment]
Is there no problem that even I close this pullrequest ??

**Comment by SandeepNayak077** [comment]
Dear Nishino-san,

No problems, you can close this pull request.

**Pull request closed without merging**